### DISCUSSION TITLE
Style to change TabStripItem margin.

### DISCUSSION THREAD

**Discussion by kevkov** [AvaloniaCommunity/Material.Avalonia#300] [Q&A] [answered]
I am trying the alter the margin for TabStripItems. The margin is defined here https://github.com/AvaloniaCommunity/Material.Avalonia/blob/fe103bda4b352b64e9a9d6c2f10382d5c0723fa7/Material.Styles/Resources/Themes/TabStripItem.axaml#L21

The style I have added, with exaggerated values so the layout effect is obvious, is
```xml
        <Style Selector="TabStripItem /template/ ContentPresenter#headerContent">
            <Setter Property="Margin" Value="50,100,100,50"></Setter>
        </Style>
```

This does not change the margin though the style is recognised in the dev tools window:
![image](https://github.com/AvaloniaCommunity/Material.Avalonia/assets/17431305/37049e0e-d77f-44b7-885e-d05413e4d621)

The applied margin is that defined by Material.Avalonia. Can someone help me define a style that changes the margin please?

**Comment by kevkov** [comment]
For context, I am displaying the items on a phone and want to reduce the space taken by a horizontal TabStrip in portrait mode.

**Comment by kevkov** [comment]
This style applies the margin correctly:
```xml
        <Style Selector="TabStripItem:not(:selected) /template/ ContentPresenter#headerContent">
            <Setter Property="Margin" Value="50,100,100,50"></Setter>
        </Style>
```
The selector is used in the original style to specify opacity. I can understand why the selector is sufficient to apply the margin but not why it is necessary.

**Comment by SKProCH** [comment, marked as answer]
It doesn't work because of priority. Since it has been set inside a template it has a `LocalValue` priority:
https://github.com/AvaloniaCommunity/Material.Avalonia/blob/fe103bda4b352b64e9a9d6c2f10382d5c0723fa7/Material.Styles/Resources/Themes/TabStripItem.axaml#L21

And your style can't override it. 

I've changed it and now you can override it. Use
```xaml
        <Style Selector="TabStripItem /template/ ContentPresenter#PART_HeaderContent">
            <Setter Property="Margin" Value="50,100,100,50"></Setter>
        </Style>
```

**Reply by kevkov** [reply]
Thanks very much. I will have to read up about this. I am fairly new Avalonia, having come via F#/Fabulous.Avalonia. Material.Avalonia is great.
